server/service/src/processors/transfer/requisition/mod.rs

```diff
@@ -0,0 +1,190 @@
+pub(crate) mod create_response_requisition;
+pub(crate) mod link_request_requisition;
+pub(crate) mod update_request_requisition_status;
+
+#[cfg(test)]
+pub(crate) mod test;
+
+use repository::{
+    ChangelogAction, ChangelogFilter, ChangelogRepository, ChangelogRow, ChangelogTableName,
+    EqualFilter, KeyValueStoreRepository, KeyValueType, RepositoryError, Requisition,
+    RequisitionFilter, RequisitionRepository, StorageConnection,
+};
+use thiserror::Error;
+
+use crate::{
+    processors::transfer::requisition::{
+        create_response_requisition::CreateResponseRequisitionProcessor,
+        link_request_requisition::LinkeRequestRequisitionProcessor,
+        update_request_requisition_status::UpdateRequestRequstionStatusProcessor,
+    },
+    service_provider::ServiceProvider,
+    sync::{ActiveStoresOnSite, GetActiveStoresOnSiteError},
+};
+
+const CHANGELOG_BATCH_SIZE: u32 = 20;
+
+#[derive(Clone, Debug)]
+pub(crate) struct RequisitionTransferProcessorRecord {
+    requisition: Requisition,
+    linked_requisition: Option<Requisition>,
+    other_party_store_id: String,
+}
+
+#[derive(Error, Debug)]
+pub(crate) enum ProcessRequisitionTransfersError {
+    #[error("Problem getting upsert record {0}")]
+    GetUpsertRecordError(GetUpsertRecordError),
+    #[error("{0}")]
+    GetActiveStoresOnSiteError(GetActiveStoresOnSiteError),
+    #[error("{0:?}")]
+    DatabaseError(RepositoryError),
+    #[error("{0}")]
+    ProcessorError(ProcessorError),
+    #[error("Name id is missing from requisition changelog {0:?}")]
+    NameIdIsMissingFromChangelog(ChangelogRow),
+    #[error("Name is not an active store {0:?}")]
+    NameIsNotAnAciveStore(ChangelogRow),
+}
+
+// TODO transaction
```

Comment:
Feels like some kind of transactions should be used in this process. At least where we are creating requisitions and requision_lines for example. We don't want to necessarily roll back on every error though so not sure.
Figure it's usually useful to comment on any 'todos' in a PR review anyway just to make sure it's been considered.

Reply:
Ah, I see the we have a transaction wrapping
`try_process_record_common` so maybe that covers off this requirement?

Reply:
Yep, I just left a comment to make sure i dont' forget, but added it a few times and forgot to delete this comment: cdf1c83ffd1f795961007396268e33155aa798e8